modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/display/context/HeaderFragmentDisplayContext.java

```diff
@@ -0,0 +1,509 @@
+
+		_themeDisplay = _cpRequestHelper.getThemeDisplay();
+
+		_commerceContext = (CommerceContext)httpServletRequest.getAttribute(
+			CommerceWebKeys.COMMERCE_CONTEXT);
+
+		_accountEntry = _commerceContext.getAccountEntry();
+	}
+
+	public CommerceChannel fetchCommerceChannel() {
+		return _commerceChannelLocalService.fetchCommerceChannelBySiteGroupId(
+			_cpRequestHelper.getScopeGroupId());
+	}
+
+	public AccountEntry getAccountEntry() {
+		return _accountEntry;
+	}
+
+	public String getCheckoutOrderURL() throws Exception {
+		CommerceOrder commerceOrder = getCommerceOrder();
+
+		if (ListUtil.exists(
+				commerceOrder.getCommerceOrderItems(),
+				CommerceOrderItemModel::isPriceOnApplication)) {
+
+			return PortletURLBuilder.create(
+				_commerceOrderHttpHelper.getCommerceCartPortletURL(
+					_httpServletRequest, commerceOrder)
+			).toString();
+		}
+
+		return PortletURLBuilder.create(
+			_commerceOrderHttpHelper.getCommerceCheckoutPortletURL(
+				_httpServletRequest)
+		).setParameter(
+			"commerceOrderId", commerceOrder.getCommerceOrderId()
+		).buildString();
+	}
+
+	public long getCommerceAccountId() {
+		long accountEntryId = 0;
+
+		if (_accountEntry != null) {
+			accountEntryId = _accountEntry.getAccountEntryId();
+		}
+
+		return accountEntryId;
+	}
+
+	public CommerceOrder getCommerceOrder() throws PortalException {
+		long commerceOrderId = getCommerceOrderId();
+
+		if (commerceOrderId > 0) {
+			return _commerceOrderService.fetchCommerceOrder(
+				getCommerceOrderId());
+		}
+
+		return _commerceOrderService.fetchCommerceOrder(
+			ParamUtil.getString(_httpServletRequest, "commerceOrderUuid"),
+			_cpRequestHelper.getCommerceChannelGroupId());
+	}
+
+	public long getCommerceOrderId() {
+		return ParamUtil.getLongValues(
+			_httpServletRequest,
+			"_com_liferay_commerce_order_content_web_internal_portlet_" +
+				"CommerceOpenOrderContentPortlet_commerceOrderId")[0];
+	}
+
+	public List<HeaderActionModel> getHeaderActionModels() throws Exception {
+		List<HeaderActionModel> headerActionModels = new ArrayList<>();
+
+		CommerceOrder commerceOrder = getCommerceOrder();
+
+		if (isShowRetryPayment()) {
+			headerActionModels.add(
+				new HeaderActionModel(
+					"btn-primary", null, getRetryPaymentURL(), null,
+					"retry-payment"));
+		}
+
+		CommerceOrderStatus currentCommerceOrderStatus =
+			_commerceOrderEngine.getCurrentCommerceOrderStatus(commerceOrder);
+
+		if ((currentCommerceOrderStatus == null) ||
+			!currentCommerceOrderStatus.isComplete(commerceOrder) ||
+			(currentCommerceOrderStatus.getKey() ==
+				CommerceOrderConstants.ORDER_STATUS_CANCELLED) ||
+			(currentCommerceOrderStatus.getKey() ==
+				CommerceOrderConstants.ORDER_STATUS_IN_PROGRESS)) {
+
+			return headerActionModels;
+		}
+
+		if (!commerceOrder.isOpen()) {
+			headerActionModels.add(
+				new HeaderActionModel(
+					"btn-primary", null, getOrderDetailURL(), null, "reorder"));
```

Comment:
If we implement the `postOrder or `postCart` for `reorder` in the JS side, we can use the current `getOrderDetailURL` implementation to confirm the behavior will work as expected.

I was not able to find Alessio's PR so I was not able to test. If we have an established URL construction for the full order details fragment view (i.e. as you mentioned before, `url/order/order_id`), then we can skip the following block from `getOrderDetailURL`:

```
	long plid = _portal.getPlidFromPortletId(
			groupId, CommercePortletKeys.COMMERCE_OPEN_ORDER_CONTENT);

		if (plid > 0) {
			portletURL = PortletProviderUtil.getPortletURL(
				_httpServletRequest, CommerceOrder.class.getName(),
				PortletProvider.Action.EDIT);

			if (getCommerceOrder() != null) {
				portletURL.setParameter(
					"mvcRenderCommandName",
					"/commerce_open_order_content/edit_commerce_order");
				portletURL.setParameter(
					"commerceOrderId", String.valueOf(getCommerceOrderId()));

				String backURL = ParamUtil.getString(
					_httpServletRequest, "backURL");

				portletURL.setParameter("backURL", backURL);
			}

			return portletURL.toString();
		}


```
and return the established URL pattern as the URL.